### PR TITLE
[ entropy_src, csrng, top_earlgrey, dv ] Add Known answer tests ...

### PR DESCRIPTION
**Pull request by martin-lueker**
…to testplan

Adds new top-level tests for entropy_src and csrng to ensure that it is possible to run the
known-answer tests in firmware.

Signed-off-by: Martin Lueker-Boden <martin.lueker-boden@wdc.com>

### CODE REVIEW

**Comment by martin-lueker** [comment]
Fixes #7140

**Review by moidx — APPROVED** [review]

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -1804,6 +1815,18 @@
       milestone: V2
       tests: []
     }
+    {
+      name: chip_sw_csrng_known_answer_tests
+      desc: '''Verify our ability to run known-answer tests in SW.
+
+            - Configure the software instance with the expected seed
+              (as per the NIST-specified test for CTR_DRBG operation)
+            - Perform several generate operations
```
Comment (mwbranstad) [inline hw/top_earlgrey/data/chip_testplan.hjson:1824]: Add a bit more detail, like which class of NIST vector tests (no_reseed, pr_false, pr_true), and how many of the examples will be run. Remove "Perform several generate operations", but rather follow the exact sequence as outlined by each vector test.

**Review by mwbranstad — APPROVED** [review]
Request for more csrng kat details, but otherwise LGTM.

**Comment by zi-v** [comment]
The following might not be directly related to the KAT, but while looking at this,I saw that 'chip_sw_entropy_src_fuse_en_fw_read' test verifies buffer read by SW. However, I am not sure it covers all that is needed to verify SW based health tests. Please consider covering the following sequence (if not already covered):
- Configure the entropy_src to FIPS mode
- Run 4096 bits through the entropy_src in FIPS mode while fw_read fuse is enabled and read the 4096 bits from the FIFO by SW
- compare the read data and make sure that all data was read correctly (note that the fifo only contains 2048 bits)
- Restart the entropy_src (with the same configuration as before) and inject the data read by SW into it.
- Verify that the entropy_src output is the same in both SW and HW cases above.

**Comment by msfschaffner** [comment]
@martin-lueker what is the status of this PR? Does this need more discussion to close?

**Comment by martin-lueker** [comment]
> @martin-lueker what is the status of this PR? Does this need more discussion to close?

Hi Michael and all,

> The following might not be directly related to the KAT, but while looking at this,I saw that 'chip_sw_entropy_src_fuse_en_fw_read' test verifies buffer read by SW. However, I am not sure it covers all that is needed to verify SW based health tests. Please consider covering the following sequence (if not already covered):
> 
> * Configure the entropy_src to FIPS mode
> * Run 4096 bits through the entropy_src in FIPS mode while fw_read fuse is enabled and read the 4096 bits from the FIFO by SW
> * compare the read data and make sure that all data was read correctly (note that the fifo only contains 2048 bits)
> * Restart the entropy_src (with the same configuration as before) and inject the data read by SW into it.
> * Verify that the entropy_src output is the same in both SW and HW cases above.

Hi @zi-v, is there anything in here that is not covered in the unit level tests?

**Comment by zi-v** [comment]
@martin-lueker , I couldn't tell for sure if all is covered from looking at the test list. The scenarios I had in mind are more for chip-level.
Anyway, I see that some of the coverage I was seeking is discussed in https://github.com/lowRISC/opentitan/issues/10983

**Comment by moidx** [comment]
Submitted this PR so that we can keep track of the KAT tests. More details can be added in follow up changes.

**Pull request merged**